production/helm/loki/templates/validate.yaml

```diff
@@ -14,3 +14,11 @@
 {{- fail "Helm test requires a prometheusAddress for an instance scraping the Loki canary's metrics"}}
 {{- end }}
 
+{{- if and (eq (include "loki.isUsingObjectStorage" .) "false") (gt (int .Values.singleBinary.replicas) 1) }}
+{{- fail "Cannot run more than 1 Single Binary replica without an object storage backend."}}
+{{- end }}
+
+{{- if and (eq (include "loki.isUsingObjectStorage" .) "false") (or (gt (int .Values.backend.replicas) 0) (gt (int .Values.read.replicas) 0) (gt (int .Values.write.replicas) 0)) }}
```

Comment:
One (potentially negative?) side effect of this check is that you have to explicitly scale down the `backend`, `read`, and `write` replicas if you're using single binary with filesystem storage. If you don't, then this check will fail.

I wonder if we need to also `and` in something like `(eq (int .Values.singleBinary.replicas) 0)` in here, like we do when defining `loki.deployment.isScalable`

Reply:
Good catch, added that extra check to the condition.